File: 102-BinaryTreeLevelOrderTraversal/102-BinaryTreeLevelOrderTraversal.java
// Last updated: 12/3/2026, 11:04:05 am
1/**
2 * Definition for a binary tree node.
3 * public class TreeNode {
4 *     int val;
5 *     TreeNode left;
6 *     TreeNode right;
7 *     TreeNode() {}
8 *     TreeNode(int val) { this.val = val; }
9 *     TreeNode(int val, TreeNode left, TreeNode right) {
10 *         this.val = val;
11 *         this.left = left;
12 *         this.right = right;
13 *     }
14 * }
15 */
16class Solution {
17    public List<List<Integer>> levelOrder(TreeNode root) {
18        List<List<Integer>> list = new ArrayList<>() ;
19        if( root==null) return list ;
20        Queue<TreeNode> q = new LinkedList<>() ;
21        q.add(root) ;
22        while( !q.isEmpty() ){
23            int size = q.size() ;
24            List<Integer> arr = new ArrayList<>() ;
25            for( int i=0 ;i<size ; i++ ) {
26                TreeNode curr = q.poll() ;
27                arr.add(curr.val) ;
28                if( curr.left!=null){
29                    q.add(curr.left);
30                }
31                if( curr.right!=null){
32                    q.add(curr.right);
33                }
34            }
35            list.add(arr);
36        }
37        return list ;
38    }
39}

File: 102-BinaryTreeLevelOrderTraversal/102-BinaryTreeLevelOrderTraversal.cpp
// Last updated: 5/8/2026, 12:09:36 am
1/**
2 * Definition for a binary tree node.
3 * struct TreeNode {
4 *     int val;
5 *     TreeNode *left;
6 *     TreeNode *right;
7 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
8 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
9 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
10 * };
11 */
12class Solution {
13public:
14    vector<vector<int>> levelOrder(TreeNode* root) {
15        vector<vector<int>> list;
16
17        if (root == nullptr)
18            return list;
19
20        queue<TreeNode*> q;
21        q.push(root);
22
23        while (!q.empty()) {
24            int size = q.size();
25            vector<int> res;
26
27            for (int i = 0; i < size; i++) {
28                TreeNode* node = q.front();
29                q.pop();
30
31                res.push_back(node->val);
32
33                if (node->left != nullptr)
34                    q.push(node->left);
35
36                if (node->right != nullptr)
37                    q.push(node->right);
38            }
39
40            list.push_back(res);
41        }
42
43        return list;
44    }
45};